amazon-q/main.tf

```diff
@@ -0,0 +1,294 @@
+        "coder": {
+          "command": "coder",
+          "args": ["exp", "mcp", "server", "--allowed-tools", "coder_report_task"],
+          "env": {
+            "CODER_MCP_APP_STATUS_SLUG": "amazon-q"
+          }
+        }
+      }
+    }
+  EOT
+  encoded_mcp_json = base64encode(local.mcp_json)
+}
+
+
+
+# Install and Initialize Amazon Q
+resource "coder_script" "amazon_q" {
+  agent_id     = var.agent_id
+  display_name = "Amazon Q"
+  icon         = var.icon
+  script       = <<-EOT
+    #!/bin/bash
+    set -o errexit
+    set -o pipefail
+
+    # Function to check if a command exists
+    command_exists() {
+      command -v "$1" >/dev/null 2>&1
+    }
+
+    # Run pre-install script if provided
+    if [ -n "${local.encoded_pre_install_script}" ]; then
+      echo "Running pre-install script..."
+      echo "${local.encoded_pre_install_script}" | base64 -d > /tmp/pre_install.sh
+      chmod +x /tmp/pre_install.sh
+      /tmp/pre_install.sh
+    fi
+
+    # Install Amazon Q if enabled
+    if [ "${var.install_amazon_q}" = "true" ]; then
+      echo "Installing Amazon Q..."
+      PREV_DIR="$PWD"
+      TMP_DIR="$(mktemp -d)"
+      cd "$TMP_DIR"
+      curl --proto '=https' --tlsv1.2 -sSf "https://desktop-release.q.us-east-1.amazonaws.com/${var.amazon_q_version}/q-x86_64-linux.zip" -o "q.zip"
+      unzip q.zip
+      ./q/install.sh --no-confirm
+      cd "$PREV_DIR"
+      export PATH="$PATH:$HOME/.local/bin"
+      echo "Installed Amazon Q version: $(q --version)"
+    fi
+
+    echo "Extracting auth tarball..."
+    PREV_DIR="$PWD"
+    echo "${var.experiment_auth_tarball}" | base64 -d > /tmp/auth.tar.zst
+    rm -rf ~/.local/share/amazon-q
+    mkdir -p ~/.local/share/amazon-q
+    cd ~/.local/share/amazon-q
+    tar -I zstd -xf /tmp/auth.tar.zst
+    rm /tmp/auth.tar.zst
+    cd "$PREV_DIR"
+    echo "Extracted auth tarball"
+
+    # Run post-install script if provided
+    if [ -n "${local.encoded_post_install_script}" ]; then
+      echo "Running post-install script..."
+      echo "${local.encoded_post_install_script}" | base64 -d > /tmp/post_install.sh
+      chmod +x /tmp/post_install.sh
+      /tmp/post_install.sh
+    fi
+
+    if [ "${var.experiment_report_tasks}" = "true" ]; then
+      echo "Configuring Amazon Q to report tasks via Coder MCP..."      
+      mkdir -p ~/.aws/amazonq
+      echo "${local.encoded_mcp_json}" | base64 -d > ~/.aws/amazonq/mcp.json
+      echo "Created the ~/.aws/amazonq/mcp.json configuration file"
+    fi
+
+    # Handle terminal multiplexer selection (tmux or screen)
+    if [ "${var.experiment_use_tmux}" = "true" ] && [ "${var.experiment_use_screen}" = "true" ]; then
+      echo "Error: Both experiment_use_tmux and experiment_use_screen cannot be true simultaneously."
+      echo "Please set only one of them to true."
+      exit 1
+    fi
+
+    FULL_PROMPT="$(printf "%s\n\nYour first task is:\n\n%s" "$CODER_MCP_AMAZON_Q_SYSTEM_PROMPT" "$CODER_MCP_AMAZON_Q_TASK_PROMPT")"
```

Comment:
Task prompt and system prompt should be configurable via module variables, and we should ship a default system prompt that instructs the agent how to do task reporting.